src/transaction_builder.js

```diff
@@ -238,14 +238,12 @@ TransactionBuilder.prototype.__build = function (allowIncomplete) {
         case 'multisig': {
           // Array.prototype.map is sparse-compatible
           var msSignatures = input.signatures.map(function (signature) {
-            return signature.toScriptSignature(input.hashType)
+            return signature && signature.toScriptSignature(input.hashType) || ops.OP_0
           })
 
-          // fill in blanks with OP_0
-          for (var i = 0; i < msSignatures.length; ++i) {
-            if (msSignatures[i]) continue
-
-            msSignatures[i] = ops.OP_0
+          // filter out OP_0s when we're not building an incomplete TX
+          if (!allowIncomplete) {
+            msSignatures = msSignatures.filter(function (signature) { return !!signature })
           }
```

Comment:
Identical to https://github.com/bitcoinjs/bitcoinjs-lib/commit/bcf8d0177dfb9b7c859b010aec695293372bf764, except isolated.